src/Mod/TechDraw/Gui/QGIViewDimension.cpp

```diff
@@ -351,21 +351,26 @@ void QGIDatumLabel::hoverLeaveEvent(QGraphicsSceneHoverEvent* event)
 
 QRectF QGIDatumLabel::boundingRect() const
 {
-    return childrenBoundingRect();
+    QRectF totalRect;
+    for (QGraphicsItem* item : m_textItems->childItems()) {
+        if (auto* customText = dynamic_cast<QGCustomText*>(item)) {
```

Comment:
You probably know this better than me, so maybe this doesn't make sense.

As I understand this, you set and clear parent items to avoid them getting added to the totalRect here in this loop. And the reason is that even an empty string has an alignmentRect which isn't 0.

To me, it looks like we're now tracking the same thing three times:
* we clear the string
* we hide the element
* and now set the parentItem to nullptr

Can't we just add a check here to see if it is visible, or has a text?
something like:
```suggestion
    for (QGraphicsItem* item : m_textItems->childItems()) {
        auto* customText = dynamic_cast<QGCustomText*>(item);
        if (customText && customText->isVisible()) {
```

Note, this is not a directive, if you don't agree, then explain to me where I'm wrong :)

Reply:
I tried isVisible at first. It didn't work. I didn't look to deeply into it, but I think there is some pre-rendering going on where. I've updated the PR to check if there is text in the custom view, this seems to work.